### DISCUSSION TITLE
new fresh install(with lazyvim) on windows hangs at Treesitter compilation step with zig

### DISCUSSION THREAD

**Discussion by seccentral** [nvim-treesitter/nvim-treesitter#6546] [Q&A]
### Describe the bug

I'm new to nvim so i did a lot of testing and distro experimenting. Settled on lazyvim but nuked my conf often to just start from scratch and learn or just redo everything because I messed it up completely.
So I know for a fact that this used to work just fine and is a recent problem (1 week or so at most)

Here's how nvim looks like with all compile tasks just taking space :
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/3766320/a7202344-f7bd-4154-81aa-00382c45abaf)
here's the details on one of them :
zig c++ -o parser.so src/parser.c -lc -Isrc -shared -Os
Current directory: "C:\Users\User\AppData\Local\nvim-data\tree-sitter-json" (this is just one, others are obviously stuck within their own respective workding directories, so this is not a "tree-sitter-json" problem, that just happens to be the current thing.

### To Reproduce

Install nvim, install zig, add it to path, git clone lazygit, launch nvim, observe the intial setup:

notice that for as long as the toast notification in the upper right corner stays active, the amount of zig compile taks in the background run. The moment that notification dissapears (or because it dissapears, who knows) nothing else happens and the list of processes just show a bunch of zigs sitting idly. They never complete the initial build.

This also happens on normal updates which need to trigger multiple zig recompiles.

This is also workaround(ed) by closing nvim, starting it again and repeating that for as many times as it takes since each run manages to finish compiling one or two tasks. (6-7 runs is enough on my machine)

### Expected behavior

The compile step should run through all taks successfuly and display a completion notification starting with 1/n up to n/n and there should be no zig processes idling in the background

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- WARNING `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
- OK `node` found v22.0.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `zig` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: info: Usage: zig [command] [options]
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "10.0.22631",
  sysname = "Windows_NT",
  version = "Windows 11 Pro N"
} ~

Parser/Features         H L F I J
  - lua                 ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.9.5
Build type: RelWithDebInfo
LuaJIT 2.1.1703942320
Compilation: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe /MD /Zi /O2 /Ob1  -W3 -wd4311 -wd4146 -DUNIT_TESTING -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE -D_WIN32_WINNT=0x0602 -DMSWIN -DINCLUDE_GENERATED_DECLARATIONS -ID:/a/neovim/neovim/.deps/usr/include/luajit-2.1 -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/build/src/nvim/auto -ID:/a/neovim/neovim/build/include -ID:/a/neovim/neovim/build/cmake.config -ID:/a/neovim/neovim/src -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/.deps/usr/include -ID:/a/neovim/neovim/.deps/usr/include

   system vimrc file: "$VIM\sysinit.vim"
  fall-back for $VIM: "C:/Program Files (x86)/nvim/share/nvim"

Run :checkhealth for more info
```

### Additional context

that's about it, i think.

**Comment by clason** [comment]
Sorry, but we can't support lazyvim setups. Please try to create a minimal config (using the script provided) someone can test with `nvim --clean -u test.lua`.

**Comment by ghost** [comment]
I had this same issue. Seem to be something with Zig. I installed gcc ("gccgo14") and it worked.
